spec/restful_resource/http_client_spec.rb

```diff
@@ -16,175 +16,98 @@ def http_client(connection)
   end
 
   describe 'Basic HTTP' do
```

Comment:
I refactored the tests, but they are the same.
I'm testing ALL the verbs with ALL the errors that we handle, not just a few.